bootstrap/016_qtag_helpers.sql

```diff
@@ -0,0 +1,95 @@
+
+-- turn a qtag into a map of maps
+create or replace function tools.qtag_to_map(qtags variant)
+returns variant
+language javascript
+as
+$$
+let ret = {};
+try {
+  QTAGS.forEach((qtag) => {
+    if (!(qtag['SOURCE'] in ret)) {
+      ret[qtag['SOURCE']] = {};
+    }
+    ret[qtag['SOURCE']][qtag.KEY] = qtag.VALUE
+  });
+  return ret;
+} catch {
+ return undefined;
```

Comment:
Would `return {}` be less onerous to deal with from the caller's side?

Reply:
Most (all) of snowflakes functions handle `NULL` as an input and return `NULL` as an output. I think the default handling here is easier than checking for an empty variant

Reply:
> Most (all) of snowflakes functions handle NULL as an input and return NULL as an output

Cool! That's good to know. I wasn't sure how the js->sql interaction worked in this case (and 30s of perusing of docs didn't lead me to the answer). Agree with your approach.